File: src/Service/adviceService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/advice';

export const getAllAdvice = () => {
  return axios.get(`${API_URL}/all`);
};

export const saveAdvice = (adviceDTO) => {
  return axios.post(`${API_URL}/saveAdvice`, adviceDTO);
};

export const deleteAdvice = (id) => {
  return axios.delete(`${API_URL}/delete/${id}`);
};

export const updateAdvice = (id, updatedAdvice) => {
  return axios.put(`${API_URL}/updateAdviceByID/${id}`, updatedAdvice);
};

export const getAdviceById = (id, getAviceById) => {
  return axios.get(`${API_URL}/getAdviceByID/${id}`, getAdviceById);
};

export const getAdviceByBmiStatus = (bmiStatus) => {
  return axios.get(`/getAdviceByStatus/${bmiStatus}`, getAdviceByBmiStatus);
};

File: GuardBackend/GuardBackend/GuardBackend/src/main/java/com/example/GuardBackend/ServiceImplementation/AdviceService.java
package com.example.GuardBackend.ServiceImplementation;

import com.example.GuardBackend.DTO.AdviceDTO;
import com.example.GuardBackend.Entity.Advice;

import java.util.List;

public interface AdviceService {

    // Save a new advice
    void saveAdvice(AdviceDTO adviceDTO);

    // Retrieve all advices
    List<Advice> getAllAdvice();

    // Delete advice by ID
    void deleteAdvice(Integer id);

    AdviceDTO updateAdviceById(Integer id, AdviceDTO adviceDTO);


    AdviceDTO getById(Integer id);

    AdviceDTO getByBmiStatus(String bmiStatus);
}
